python/google/protobuf/pyext/descriptor.cc

```diff
@@ -58,6 +58,43 @@
               : 0)                                               \
        : PyBytes_AsStringAndSize(ob, (charpp), (sizep)))
 
+#if PY_VERSION_HEX < 0x030900B1 && !defined(PYPY_VERSION)
+static inline PyCodeObject* PyFrame_GetCode(PyFrameObject *frame)
+{
+    Py_INCREF(frame->f_code);
+    return frame->f_code;
+}
+
+static inline PyFrameObject* PyFrame_GetBack(PyFrameObject *frame)
+{
+    Py_XINCREF(frame->f_back);
+    return frame->f_back;
+}
+#endif
+
+#if PY_VERSION_HEX < 0x030B00A7 && !defined(PYPY_VERSION)
+static inline PyObject* PyFrame_GetLocals(PyFrameObject *frame)
+{
+#if PY_VERSION_HEX >= 0x030400B1
+    if (PyFrame_FastToLocalsWithError(frame) < 0) {
+        return NULL;
+    }
+#else
+    PyFrame_FastToLocals(frame);
+#endif
+    Py_INCREF(frame->f_locals);
```

Comment:
I am worried that these increfs don't have corresponding decrefs.  The old code had neither and the new code has only one.

Reply:
Yes, you're right, I'll fix it

Reply:
I believe the `Py_INCREF` here is to make this function match the version of `PyFrame_GetLocals` in Python 3.11 which returns a string reference. The Python API changed to provide an few functions instead of accessing `frame->f_locals` directly.